src/components/ConversationSettings/MentionsSettings.vue

```diff
@@ -0,0 +1,156 @@
+<!--
+  - SPDX-FileCopyrightText: 2020 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<template>
+	<div v-if="canModerate">
+		<NcCheckboxRadioSwitch :checked="mentionPermissions === MENTION_PERMISSIONS.EVERYONE"
+			:disabled="isMentionPermissionsLoading"
+			type="switch"
+			@update:checked="toggleMentionPermissions">
+			{{ t('spreed', 'Allow participants to mention @all') }}
+		</NcCheckboxRadioSwitch>
+	</div>
+
+	<div v-else>
+		<h5 class="app-settings-section__subtitle">
+			{{ t('spreed', 'Mention permissions') }}
+		</h5>
+		<p>{{ summaryLabel }}</p>
+	</div>
+</template>
+
+<script>
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import { t } from '@nextcloud/l10n'
+
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+
+import { CONVERSATION } from '../../constants.js'
+
+export default {
+	name: 'MentionsSettings',
+
+	components: {
+		NcCheckboxRadioSwitch,
+	},
+
+	props: {
+		token: {
+			type: String,
+			default: null,
+		},
+
+		canModerate: {
+			type: Boolean,
+			default: true,
+		},
+
+		value: {
+			type: Number,
+			default: null,
+		},
```

Comment:
You didn't use this prop.

Reply:
Thanks for the review, frontend was just copy paste from another file for now to quickly test it but with the review it will be quick to fix now :)